### PR TITLE
record_size_limit_extension and error message.

### PR DESCRIPTION
**Pull request by erikcorry**
In client mode, always send the record_size_limit
extension to try to persuade the server to avoid large
record that we can't decode.

If the server nevertheless sends us a record that is too
large, then return an error message that explains the
situation to the programmer.

### CODE REVIEW

File: library/error.c [hunk 1]
```diff
@@ -553,6 +553,8 @@ void mbedtls_strerror( int ret, char *buf, size_t buflen )
             mbedtls_snprintf( buf, buflen, "SSL - Internal-only message signaling that a message arrived early" );
         if( use_ret == -(MBEDTLS_ERR_SSL_CRYPTO_IN_PROGRESS) )
             mbedtls_snprintf( buf, buflen, "SSL - A cryptographic operation is in progress. Try again later" );
+        if( use_ret == -(MBEDTLS_ERR_SSL_OVERSIZED_RECORD) )
+            mbedtls_snprintf( buf, buflen, "SSL - Peer sent us a record too large for our puny buffer" );
```
Comment (floitsch) [inline library/error.c]:
```suggestion
            mbedtls_snprintf( buf, buflen, "SSL - Peer sent us a record too large for our buffer" );
```

File: library/ssl_cli.c [hunk 2]
```diff
@@ -531,6 +531,46 @@ static int ssl_write_truncated_hmac_ext( mbedtls_ssl_context *ssl,
...
+    int limit = MBEDTLS_SSL_IN_CONTENT_LEN;
+    /* Set the limit a little lower than our actual buffer size to account for
+       unknown overhead.  */
+    limit -= 256;
+    limit *= 0.95;
+    limit &= ~255;
+
+    *olen = 0;
+
+    if( ssl->conf->record_size_limit == MBEDTLS_SSL_RECORD_SIZE_LIMIT_DISABLED )
```
Comment (floitsch) [inline library/ssl_cli.c:551]: Why is this not at the top of the function?

**Review by floitsch — APPROVED** [review]
LGTM.

File: library/ssl_cli.c (hunk 2 above)
Comment (erikcorry) [reply library/ssl_cli.c:551]: Part of the reason is that this is C90 and we can't declare limit anywhere except at the top of the block.  And the unconditional write to *olen is just following the pattern of other functions in this file.

**Pull request merged**